src/scripts/check.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+"""
+Implements the "make check" target
+
+(C) 2020 Jack Lloyd, Rene Meusel
+
+Botan is released under the Simplified BSD License (see license.txt)
+"""
+
+import os
+import sys
+import optparse
+import subprocess
+import logging
+import platform
+
+def is_macos():
+    return platform.system() == "Darwin"
+
+def is_linux():
+    return platform.system() == "Linux"
+
+
+def run_and_check(cmd_line, env=None, cwd=None):
+
+    logging.info("Starting %s", ' '.join(cmd_line))
+
+    try:
+        proc = subprocess.Popen(cmd_line, cwd=cwd, env=env)
+
+        proc.communicate()
+    except OSError as e:
+        logging.error("Executing %s failed (%s)", ' '.join(cmd_line), e)
+
+    if proc.returncode != 0:
+        logging.error("Error running %s", ' '.join(cmd_line))
+        sys.exit(1)
+
+
+def get_environment(shared_lib):
+    if not shared_lib:
+        return None
+
+    env = os.environ.copy()
+
+    if is_macos():
+        env["DYLD_LIBRARY_PATH"] = os.path.abspath(".")
+    elif is_linux():
+        env["LD_LIBRARY_PATH"] = os.path.abspath(".")
```

Comment:
Do we need more platforms here?

Reply:
Most other OSes (eg FreeBSD) also use LD_LIBRARY_PATH. Instead of trying to detect which to use, just always set both vars. Also need to handle existing value of these environments instead of overwriting them, eg if OpenSSL is in some side dir but user has adjusted with LD_LIBRARY_PATH.